### PR TITLE
Clarify DNS service discovery types language

### PR DESCRIPTION
**Pull request by sjiveson**
Improved the DNS service discovery language used as it was either incorrect (DNS addresses) or confusing. Also added note on what happens if a single hostname query returns multiple addresses.

### CODE REVIEW

File: docs/configuration/cluster_manager/cluster.rst [hunk 1]
```diff
@@ -64,15 +64,15 @@ hosts
       [{"url": "tcp://10.0.0.2:1234"}, {"url": "tcp://10.0.0.3:5678"}]
 
   strict_dns
-    Strict DNS clusters can specify any number of DNS/port combinations. All DNS addresses will be
-    resolved and grouped together to form the final cluster. For example:
+    Strict DNS clusters can specify any number of hostname:port combinations. All names will be
+    resolved using DNS and grouped together to form the final cluster. If multiple records are returned for a single name, all will be used. For example:
```
Comment (mattklein123) [inline docs/configuration/cluster_manager/cluster.rst]: nit: can you limit line to 100 cols.

**Comment by mattklein123** [comment]
Thanks for doing this. Can you also please sign CLA.

**Comment by sjiveson** [comment]
Apologies, CLA?

**Comment by mattklein123** [comment]
https://github.com/lyft/envoy/blob/master/CONTRIBUTING.md#cla

**Comment by sjiveson** [comment]
Thanks. Done.

**Comment by sjiveson** [comment]
Great, thank you.

**Pull request merged**